### ISSUE TITLE
What is the correct way to specify compiler options for target libstdc++?

### ISSUE THREAD

**Issue by forky2** [crosstool-ng/crosstool-ng#1673]
This is not so much of an issue, more of a question. I'm using the latest from `master`.

I'm trying to compile a complete toolchain for ARM, EABI5 *BE8*. The BE8 bit is what's getting me stumped.

Starting with the sample `armeb-unknown-linux-gnueabi` I can almost get there. I made the following changes...

* CT_GLIBC_EXTRA_CFLAGS="-mbe8" # This causes libc.so.6 for the target to be built as ARM, EABI5 BE8. 👍 
* CT_TARGET_CFLAGS="-mbe8" # This causes the gcc libs (such as libgcc_s.so.1) for the target to be built as ARM, EABI5 BE8. 👍 

However, I can't get libstdc++.so.6.0.29 to build with BE8.

I have dug into the gcc makefiles and the problem as I see it is, altough ct-ng correctly passes parameters like `CFLAGS_FOR_TARGET="-mbe8"` and `CXXFLAGS_FOR_TARGET="-mbe8"` to gcc's configure, the Makefile for libstdc++ *ignores* this at the link stage. `-mbe8` gets used for all of the compilation stages, but at the link stage (which is where it really matters) libtool is used and it has not been given the instruction.

From a libtool point of view, I believe the solution would be to update libstdc++'s Makefile where the `OPT_LDFLAGS` variable can be updated to include `-Wc,-mbe8`; that is, libtool would pass `-mbe8` through to xgcc when linking libstdc++.so.6.0.29.

And indeed I have manually tested that this would work, but it's an ugly hack that takes you out of the tidy ct-ng universe.

Is there a better way already supported in ct-ng? If not I'm guessing that I need to create a new mutable config option in ct-ng, altough what would be the implementation? (i.e. is there a parameter that can be passed to gcc ./configure stage to do what I want, or will I need to have the gcc scripts patched?)

Any help appreciated. Also happy to offer a build log if it's of interest, although I suspect that will be unnecessary noise for people that know the ct-ng implementation well.

**Comment by antmak** [comment]
@forky2 If I understand correct, please see `CC_GCC_ENABLE_CXX_FLAGS`. `Enter here the value of the gcc's ./configure option --enable-cxx-flags.`  and about `--enable-cxx-flags` here:  https://gcc.gnu.org/onlinedocs/libstdc++/manual/configure.html

**Comment by antmak** [comment]
>  If I set CT_GLIBC_EXTRA_CFLAGS="-mbe8", I still end up with a non-BE8 libstdc++.so.6.0.29.

I meant `CC_GCC_ENABLE_CXX_FLAGS`, not `CT_GLIBC_EXTRA_CFLAGS` :) That didn't work too?

**Comment by forky2** [comment]
I apologise, I did enable the one you mentioned, but they I retold how another one I also have set didn't work. So to clarify, with both `CC_GCC_ENABLE_CXX_FLAGS` and `GLIBC_EXTRA_CFLAGS` set to "-mbe8" it is still built non-BE8. (Also, all options have "CT_" prefixed in the config, correct?)

**Comment by antmak** [comment]
yes, `CT_` prefix is required. You can try to set the variable via `./ct-ng menuconfig`, also I guess to check help strings for another options in menuconfig... Sorry, no more ideas now about your problem
